From a6b23e8a7b9ac3f6a95b522833400d4365c25fa2 Mon Sep 17 00:00:00 2001
From: Johannes Lorenz <j.git@lorenz-ho.me>
Date: Sun, 29 Mar 2020 11:08:56 +0200
Subject: [PATCH 1508/1936] Fix PAD filter changes needing prepare

Before this, any realtime parameter change of PAD synth, e.g. filter,
triggered the "needPrepare" message, marking the PAD synth's "Apply
Changes" button.
---
 src/Misc/MiddleWare.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Misc/MiddleWare.cpp b/src/Misc/MiddleWare.cpp
index 12e3ad34..9d0cbdee 100644
--- a/src/Misc/MiddleWare.cpp
+++ b/src/Misc/MiddleWare.cpp
@@ -322,7 +322,7 @@ struct NonRtObjStore
                 strcpy(d.loc, obj_rl.c_str());
                 d.obj = pad;
                 PADnoteParameters::non_realtime_ports.dispatch(msg, d);
-                if(rtosc_narguments(msg)) {
+                if(d.matches && rtosc_narguments(msg)) {
                     if(!strcmp(msg, "oscilgen/prepare"))
                         ; //ignore
                     else {
-- 
2.47.0

